src/github/prComment.ts

```diff
@@ -0,0 +1,183 @@
+	/**
+	 * The id of the comment
+	 */
+	public id: number;
+
+	/**
+	 * The comment thread the comment is from
+	 */
+	public parent: GHPRCommentThread;
+
+	/**
+	 * The text of the comment
+	 */
+	public body: string | vscode.MarkdownString;
+
+	/**
+	 * If the temporary comment is in place for an edit, the original text value of the comment
+	 */
+	public originalBody?: string;
+
+	/**
+	 * Whether the comment is in edit mode or not
+	 */
+	public mode: vscode.CommentMode;
+
+	/**
+	 * The author of the comment
+	 */
+	public author: vscode.CommentAuthorInformation;
+
+	/**
+	 * The label to display on the comment, 'Pending' or nothing
+	 */
+	public label: string | undefined;
+
+	/**
+	 * The list of reactions to the comment
+	 */
+	public commentReactions?: vscode.CommentReaction[] | undefined;
+
+	/**
+	 * The context value, used to determine whether the command should be visible/enabled based on clauses in package.json
+	 */
+	public contextValue: string;
+
+	static idPool = 0;
+
+	constructor(parent: GHPRCommentThread, input: string, isDraft: boolean, currentUser: IAccount, originalComment?: GHPRComment) {
+		this.parent = parent;
+		this.body = new vscode.MarkdownString(input);
+		this.mode = vscode.CommentMode.Preview;
+		this.author = {
+			name: currentUser.login,
+			iconPath: currentUser.avatarUrl ? vscode.Uri.parse(`${currentUser.avatarUrl}&s=${64}`) : undefined
+		};
+		this.label = isDraft ? 'Pending' : undefined;
+		this.contextValue = 'canEdit,canDelete';
+		this.originalBody = originalComment ? originalComment._rawComment.body : undefined;
+		this.commentReactions = originalComment ? originalComment.reactions : undefined;
+		this.id = TemporaryComment.idPool++;
+	}
+}
+
+export class GHPRComment implements vscode.Comment {
+	/**
+	 * The database id of the comment
+	 */
+	public commentId: string;
+
+	/**
+	 * The comment thread the comment is from
+	 */
+	public parent: GHPRCommentThread;
+
+	/**
+	 * The text of the comment
+	 */
+	public body: string | vscode.MarkdownString;
+
+	/**
+	 * Whether the comment is in edit mode or not
+	 */
+	public mode: vscode.CommentMode;
+
+	/**
+	 * The author of the comment
+	 */
+	public author: vscode.CommentAuthorInformation;
+
+	/**
+	 * The label to display on the comment, 'Pending' or nothing
+	 */
+	public label: string | undefined;
+
+	/**
+	 * The list of reactions to the comment
+	 */
+	public reactions?: vscode.CommentReaction[] | undefined;
+
+	/**
+	 * The complete comment data returned from GitHub
+	 */
+	public _rawComment: IComment;
+
+	/**
+	 * The context value, used to determine whether the command should be visible/enabled based on clauses in package.json
+	 */
+	public contextValue: string;
+
+	constructor(comment: IComment, parent: GHPRCommentThread) {
+		this._rawComment = comment;
+		this.commentId = comment.id.toString();
+		this.body = new vscode.MarkdownString(`hello [create pull request](command:pr.create)`); // comment.body);
```

Comment:
Is this hardcoded body value meant to be here?

Reply:
good catch! nope :)